src/wallet/wallet2.cpp

```diff
@@ -1267,6 +1267,11 @@ bool wallet2::has_stagenet_option(const boost::program_options::variables_map& v
   return command_line::get_arg(vm, options().stagenet);
 }
 
+bool wallet2::has_proxy_option(const boost::program_options::variables_map& vm)
+{
+  return command_line::has_arg(vm, options().proxy);
+}
```

Comment:
```suggestion
bool wallet2::has_proxy_option() const
{
  return !m_proxy.empty();
}
```

I'd recommend this approach since it reflects the value per `wallet2` instance, even if it wasn't initiated by a CLI.

Reply:
What you are suggesting makes more sense. I fixed it in a new push. Thanks.